src/databricks/labs/ucx/install.py

```diff
@@ -282,18 +284,31 @@ def _confirm_force_install(self) -> bool:
         return False
 
     def configure(self, default_config: WorkspaceConfig | None = None) -> WorkspaceConfig:
+        connection_error_message = (
+            f"Cannot connect with {self.workspace_client.config.host} see "
+            "https://github.com/databrickslabs/ucx#network-connectivity-issues for help: %s"
+        )
         try:
             config = self.installation.load(WorkspaceConfig)
             self._compare_remote_local_versions()
-            if self._confirm_force_install():
-                return self._configure_new_installation(default_config)
-            self._apply_upgrades()
-            return config
+            if not self._confirm_force_install():
+                self._apply_upgrades()
+                return config
         except NotFound as err:
             logger.debug(f"Cannot find previous installation: {err}")
         except (PermissionDenied, SerdeError, ValueError, AttributeError):
             logger.warning(f"Existing installation at {self.installation.install_folder()} is corrupted. Skipping...")
-        return self._configure_new_installation(default_config)
+        except TimeoutError as err:
+            # API calls are wrapped with a retry mechanism for requests.exceptions.ConnectionError
+            # eventually raising a TimeoutError if tries take too long
+            if isinstance(err.__cause__, RequestsConnectionError):  # raise TimeoutError(...) from ConnectionError(...)
+                logger.warning(connection_error_message, err)
```

Comment:
What is supposed to happen if a `TimeoutError` reaches here that is _not_ caused by `RequestsConnectionError`? At the moment we simply swallow it without any indication whereas previously it was propagated.

Reply:
We retry with default config, see other comment about code complexity. If it fails there again, we raise it

Reply:
Also, this was only the case if `_confirm_force_install` is `False` otherwise the flow remains the same. That those two code flows are different is also confusing